code/extensions/GridFieldBetterButtonsItemRequest.php

```diff
@@ -548,11 +548,35 @@ public function recordIsPublished() {
 
 	}
 
+	/**
+	 * Determines if the current record is deleted from stage
+	 * @returns boolean
+	 */
+	public function recordIsDeletedFromStage() {
+
+		// for SiteTree records
+		if ( $this->owner->hasMethod('getIsDeletedFromStage') ) {
+			return $this->IsDeletedFromStage;
+		}
+
+		if(!$this->owner->record->checkVersioned()) return false;
```

Comment:
I'm not sure what @unclecheese prefers but I like all if statements to have braces on them.